### ISSUE TITLE
Fix JSON API

### ISSUE THREAD

**Issue by ernestask** [abrt/faf#892]
It’s just completely nonsensical, because we:

- try to return JSON-unserializable types (https://github.com/abrt/faf/blob/5e15cfc50f363567e5f638655d5ee14fbe6bcc0a/src/webfaf/reports.py#L590, though for that you can use/extend `WebfafJSONEncoder`),
- try to return JSON-unserializable types that don’t make any sense in the context (https://github.com/abrt/faf/blob/5e15cfc50f363567e5f638655d5ee14fbe6bcc0a/src/webfaf/problems.py#L495),
- return wildly inconsistent data (`null`, `[null]` for the same field).

In short: we should have a well-defined API and not mix endpoints that return JSON or Flask-rendered HTML based on the temperature outside, because we don’t have the tests to verify the correctness (over time as well) and because there should be a clear separation between the backend and web UI, due to its optional nature.

One attempt was https://github.com/abrt/faf/pull/886, but that broke everything and I’ve since discovered that it goes deeper.
